client/ayon_core/tools/utils/delegates.py

```diff
@@ -130,44 +130,113 @@ def paint(self, painter, option, index):
+        if text:
+            option.features |= QtWidgets.QStyleOptionViewItem.HasDisplay
+            option.text = text
+
         painter.save()
+        painter.setClipRect(option.rect)
 
+        icon_rect = style.subElementRect(
+            QtWidgets.QCommonStyle.SE_ItemViewItemDecoration,
+            option,
+            option.widget
+        )
         text_rect = style.subElementRect(
             QtWidgets.QCommonStyle.SE_ItemViewItemText,
-            option
+            option,
+            option.widget
         )
-        text_margin = style.proxy().pixelMetric(
-            QtWidgets.QCommonStyle.PM_FocusFrameHMargin,
+
+        # Draw background
+        style.drawPrimitive(
+            QtWidgets.QCommonStyle.PE_PanelItemViewItem,
             option,
+            painter,
             option.widget
-        ) + 1
-        padded_text_rect = text_rect.adjusted(
-            text_margin, 0, - text_margin, 0
         )
 
+        # Draw icon
+        option.icon.paint(
+            painter,
+            icon_rect,
+            option.decorationAlignment,
+            mode,
+            state
+        )
         fm = QtGui.QFontMetrics(option.font)
-        text = self._get_status_name(index)
-        if padded_text_rect.width() < fm.width(text):
+        if text_rect.width() < fm.width(text):
             text = self._get_status_short_name(index)
+            if text_rect.width() < fm.width(text):
+                text = ""
 
         fg_color = self._get_status_color(index)
         pen = painter.pen()
         pen.setColor(fg_color)
         painter.setPen(pen)
 
         painter.drawText(
-            padded_text_rect,
+            text_rect,
             option.displayAlignment,
             text
         )
 
+        if option.state & QtWidgets.QStyle.State_HasFocus:
+            focus_opt = QtWidgets.QStyleOptionFocusRect()
+            focus_opt.state = option.state
+            focus_opt.direction = option.direction
+            focus_opt.rect = option.rect
+            focus_opt.fontMetrics = option.fontMetrics
+            focus_opt.palette = option.palette
+
+            focus_opt.rect = style.subElementRect(
+                QtWidgets.QCommonStyle.SE_ItemViewItemFocusRect,
+                option,
+                option.widget
+            )
+            focus_opt.state |= (
+                QtWidgets.QStyle.State_KeyboardFocusChange
+                | QtWidgets.QStyle.State_Item
+            )
+            focus_opt.backgroundColor = option.palette.color(
+                (
+                    QtWidgets.QPalette.Normal
+                    if option.state & QtWidgets.QStyle.State_Enabled
+                    else QtWidgets.QPalette.Disabled
+                ),
+                (
+                    QtWidgets.QPalette.Highlight
+                    if option.state & QtWidgets.QStyle.State_Selected
+                    else QtWidgets.QPalette.Window
+                )
+            )
```

Comment:
I believe `QPalette` is in `QtGui`?
```suggestion
            focus_opt.backgroundColor = option.palette.color(
                (
                    QtGui.QPalette.Normal
                    if option.state & QtWidgets.QStyle.State_Enabled
                    else QtGui.QPalette.Disabled
                ),
                (
                    QtGui.QPalette.Highlight
                    if option.state & QtWidgets.QStyle.State_Selected
                    else QtGui.QPalette.Window
                )
            )
```

Reply:
Fixed with https://github.com/ynput/ayon-core/pull/593/commits/4264af63aff45a03e48b2ca6f8868747a24039bc